inputfiles/overridingTypes.jsonc

```diff
@@ -2573,7 +2573,7 @@
                                 },
                                 {
                                     "name": "strategy",
-                                    "overrideType": "{ highWaterMark?: number }"
+                                    "overrideType": "{ highWaterMark?: number; size?: undefined }"
```

Comment:
Why an unused property? Who would be happy with this? Can you add a comment about that? (it's jsonc so it supports comment)

Reply:
Hmm okay, but doesn't TypeScript already show error for excessive fields?

Reply:
It's a minor detail, but we can make `new ReadableStream({ type: “bytes” }, { size: undefined })` be inferred as `ReadableStream<Uint8Array>`.

<img width="500" alt="" src="https://github.com/user-attachments/assets/4f1180e8-470e-434f-a05f-8817ce38a0ce">

https://www.typescriptlang.org/play/?#code/MYewdgzgLgBATgRhgXhmApgdxgJXQQwBN8AjAG3QGUo4CBbACgG8YoBPAB3QC4YAiEmyjoIfGAF8ANDCbiAlACgA9EphqYAPQD8ChaEiw4AJhRosuAsXJUa9Zq049+g4aInSWEAJYAvJwFcwQnQAMy8MQglFFXVNHSA

Reply:
Hmm that's interesting... Is that something frequently done by users?

Reply:
In general, I believe that implementations that set properties to `undefined` are often used because it's easy to write, while spread syntax is required to implement empty properties in some cases.

```js
const option1 = {
  foo: "foo",
  bar: setBar ? "bar" : undefined,
};

const option2 = {
  foo: "foo",
  ...(setBar ? { bar: "bar" } : {}),
};
```

Reply:
But what is the hypothetical setBar here? readAsBytes? I would expect either always pass size or never pass size here than your conditional pass example, unless the caller also has the hypothetical setBar for `type`, which would be kinda complex setup to create ReadableStream. Especially when the way to read the resulting stream would be significantly different.

Reply:
I think it is nonsense to consider whether or not to give `?: undefined` per API; it is consistent to allow for the `undefined` assignment, as in TypeScript's `Partial`. However, if you have strong concerns here, I am not opposed to revert.

Reply:
> I think it is nonsense to consider whether or not to give `?: undefined` per API; it is consistent to allow for the `undefined` assignment, as in TypeScript's `Partial`. However, if you have strong concerns here, I am not opposed to revert.

Agreed that it's nonsense to decide per API (or at least hard to maintain), and I think it should not exist for consistency sake because it would be the first case if we add this here. That is unless we have a strong proof that there's many users who do setBar-style thing here (which I think doesn't make sense at least for this one).

Reply:
> I think it should not exist for consistency sake because it would be the first case if we add this here. That is unless we have a strong proof that there's many users who do setBar-style thing here (which I think doesn't make sense at least for this one).

Agreed. I reverted this change.

Reply:
Thanks, I'll see whether #1713 is ready enough to be merged and when I can land this.